lib/rialto/etl/writers/sparql_statement_writer.rb

```diff
@@ -0,0 +1,191 @@
+# frozen_string_literal: true
+
+require 'rdf'
+require 'traject'
+require 'uuid'
+require 'sparql/client'
+require 'rialto/etl/namespaces'
+
+module Rialto
+  module Etl
+    # Holds writers for use in Traject mappings
+    module Writers
+      # rubocop:disable Metrics/ClassLength
+      # Write Sparql statement records
+      class SparqlStatementWriter < Traject::LineWriter
+        extend Rialto::Etl::Vocabs
+        # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/MethodLength
+        # Overrides the serialization routine from superclass
+        #
+        # @param context [Traject::Indexer::Context] a Traject context
+        #   object containing the output of the mapping
+        # @return [String] Sparql representation of the mapping
+        def serialize(context)
+          # serialize_hash is a separate method to allow for recursion
+          serialize_hash(context.output_hash).flatten.join(";\n") + ";\n"
+        end
+
+        def serialize_hash(hash, graph_name = nil)
+          statements = []
+          subject_id = hash['@id'].to_s
+          subject = RDF::URI.new(hash['@id_ns'] + subject_id)
+          graph_name ||= hash['@graph']
+
+          # Type
+          statements << values_to_delete_insert(subject, RDF.type, hash['@type'], graph_name, hash.key?('!type'))
+
+          # Label
+          # SKOS.prefLabel & VCARD.fn
+          statements << values_to_delete_insert(subject,
+                                                [Vocabs::SKOS['prefLabel'],
+                                                 Vocabs::FOAF['fn']], hash['@label'],
+                                                graph_name,
+                                                hash.key?('!label'))
+
+          # Person name vcard
+          statements << person_name_to_statements(subject, hash, graph_name) if hash.key?('@person_name')
+
+          # Person address vcard
+          statements << person_address_to_statements(subject, hash, graph_name) if hash.key?('@person_address')
+
+          # Advisees
+          statements << advisees_to_statements(subject, subject_id, hash, graph_name) if hash.key?('@advisees') &&
+                                                                                         !hash['@advisees'].empty?
+
+          # All other
+          statements << hash_to_delete_insert(subject, hash, graph_name)
+        end
+        # rubocop:enable Metrics/AbcSize
+        # rubocop:enable Metrics/MethodLength
+
+        def graph_to_insert(graph, graph_name)
+          SPARQL::Client::Update::InsertData.new(graph, graph: graph_name).to_s.chomp
+        end
+
+        def graph_to_delete(graph, graph_name)
+          SPARQL::Client::Update::DeleteInsert.new(graph,
+                                                   nil,
```

Comment:
Can you remind me again, why we can't have the insert graph as the second argument to `DeleteInsert`? Can you add a comment about this?

Reply:
Will add comment to the effect: Cannot perform insert in deleteinsert because insert is only performed when the where clause is satisfied. Thus, it works for an update but not an initial insert.